[prometheus-pushgateway] tolerations should be list not map

### Describe the bug a clear and concise description of what the bug is.

[values.yaml lines 167-1690](https://github.com/prometheus-community/helm-charts/blob/main/charts/prometheus-pushgateway/values.yaml#L167-L169)  `tolerations:` default is a map, but the example in comments is a list.  Tolerations in kubernetes spec API is usually a list.

When using helm, I get these warnings:
```
coalesce.go:220: warning: cannot overwrite table with non table for prometheus-pushgateway.tolerations (map[])
coalesce.go:220: warning: cannot overwrite table with non table for prometheus-pushgateway.tolerations (map[])
```


### What's your helm version?

version.BuildInfo{Version:"v3.8.0", GitCommit:"d14138609b01886f544b2025f5000351c9eb092e", GitTreeState:"clean", GoVersion:"go1.17.5"}

### What's your kubectl version?

Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.15", GitCommit:"b84cb8ab29366daa1bba65bc67f54de2f6c34848", GitTreeState:"clean", BuildDate:"2022-12-08T10:49:13Z", GoVersion:"go1.17.13", Compiler:"gc", Platform:"linux/amd64"}

### Which chart?

prometheus-pushgateway

### What's the chart version?

 2.0.4

### What happened?

When using helm, I get these warnings:
```
coalesce.go:220: warning: cannot overwrite table with non table for prometheus-pushgateway.tolerations (map[])
coalesce.go:220: warning: cannot overwrite table with non table for prometheus-pushgateway.tolerations (map[])
```

Also, the example in values.yaml for `tolerations` contradicts the default value.

### What you expected to happen?

No warnings, and a default value that matches what I expect in the Kubernetes spec API for Statefulsets

### How to reproduce it?

`helm install my-release prometheus-community/prometheus-pushgateway --values newvalues.yaml` where newvalues.yaml has the example described in values.yaml:
`tolerations: [ {effect: NoSchedule, operator: Exists} ]`

### Enter the changed values of values.yaml?

tolerations: [ {effect: NoSchedule, operator: Exists} ]

### Enter the command that you execute and failing/misfunctioning.

`helm install my-release prometheus-community/prometheus-pushgateway --values newvalues.yaml`

### Anything else we need to know?

Maybe you didn't notice because you've only tested with empty `tolerations:`  and `toYaml` an empty map makes the same output as `toYaml` an empty list.

--- a/charts/prometheus-pushgateway/Chart.yaml
+++ b/charts/prometheus-pushgateway/Chart.yaml
@@ -1,11 +1,12 @@
 apiVersion: v2
-appVersion: "v1.5.1"
+appVersion: "v1.6.0"
 description: A Helm chart for prometheus pushgateway
 name: prometheus-pushgateway
-version: 2.2.0
+version: 2.3.0
 home: https://github.com/prometheus/pushgateway
 sources:
 - https://github.com/prometheus/pushgateway
+type: application
 keywords:
 - pushgateway
 - prometheus
@@ -16,3 +17,8 @@ maintainers:
   name: cstaud
 - email: rootsandtrees@posteo.de
   name: zeritti
+annotations:
+  "artifacthub.io/license": Apache-2.0
+  "artifacthub.io/links": |
+    - name: Chart Source
+      url: https://github.com/prometheus-community/helm-charts

--- a/charts/prometheus-pushgateway/values.yaml
+++ b/charts/prometheus-pushgateway/values.yaml
@@ -164,7 +164,7 @@ ingress:
   #     hosts:
   #       - pushgateway.domain.com
 
-tolerations: {}
+tolerations: []
   # - effect: NoSchedule
   #   operator: Exists
 

